documentation/book/kafka-exporter/con-kafka-exporter-lag.adoc

```diff
@@ -1,12 +1,12 @@
-// This assembly is included in the following assemblies:
+// Module included in the following assemblies:
 //
 // assembly-kafka-exporter.adoc
-[id='con-metrics-kafka-exporter-lag{context}']
+[id='con-metrics-kafka-exporter-lag-{context}']
 
 = Consumer lag
 
 Consumer lag indicates the difference in the rate of production and consumption of messages.
-Specifically, consumer lag indicates the delay between a Kafka broker writing a message created by a producer and the rate in which it is picked up by a consumer.
+Specifically, the lag for a given message and consumer is the delay between a Kafka broker writing the message to its log and the time a consumer offset is incremented past that message offset.
```

Comment:
I do not think there is any lag for specific message. The consumer lag is the difference between the position of the consumer and the end of the partition. It also doens't show time but just an offset difference. I think the old sentence was better and less confusing than the new one.

Reply:
Can someone please explain how a lag between an event and a rate makes any sense?

Reply:
The delay is the difference in the rate of message production and consumption. We could drop 'rate' to have:

Specifically, consumer lag indicates the delay between a Kafka broker writing a message created by a producer and the message being picked up by a consumer. When a broker writes a message to its log, any delay will reflect on the position of the consumer offset in relation to the end of the log.

This difference is sometimes referred to as the delta between the producer offset and consumer offset...

Reply:
How about (to make it clear that it's the lag for a consumer):

> Specifically, consumer lag for a given consumer indicates the delay between a Kafka broker writing a message created by a producer and the message being picked up by that consumer. When a broker writes a message to its log, any delay will reflect on the position of the consumer offset in relation to the end of the log.

Reply:
It's clear to me. @scholzj?

Reply:
@tombentley I think this is still a bit confusing. for example the lag could be there there even wihtout the broker writting new messages. I think it can be also interpreted as if the lag is per message. But it is only per partition. What about:

> Specifically, consumer lag for a given consumer group indicates the delay between the last message in the partition and the message being currently picked up by that consumer. The lag reflects the position of the consumer offset in relation to the end of the partition log.

Reply:
That works for me.

Reply:
:thumbsup: (I feel tempted to add a diagram after the next release.)